### PR TITLE
[DUOS-1015][risk=no] Refactor

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1015

## Changes
Updated the following pages so that they use the `ApplicationDownloadLink` to download a DAR's PDF file instead of making an API call to consent.
* Access Collect (admin view for existing DAR election)
* Access Preview (admin view for a new DAR election)
* Access Results Record (admin view for Statistics -> Reviewed Cases Record -> Record)

Updated the common component, `ApplicationSummary` with some bug fixes where we were not looking at the correct researcher properties for display.

Discovered a bug where we were showing the wrong machine readable DAR content. Filed in [DUOS-1089](https://broadworkbench.atlassian.net/browse/DUOS-1089)

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] PR is labeled with a Jira ticket number and includes a link to the ticket
- [ ] PR is labeled with a security risk modifier [no, low, medium, high] 
- [ ] PR describes scope of changes

In all cases:

- [ ] Get a minimum of one thumbs worth of review, preferably 2 if enough team members are available
- [ ] Get PO sign-off for all non-trivial UI or workflow changes
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/components/ApplicationSummary.js [hunk 1]
```diff
@@ -23,14 +24,14 @@ export const ApplicationSummary = hh(class ApplicationSummary extends PureCompon
       div({ id: 'panel_applicationSummary', className: 'panel-body row' }, [
         div({ className: 'col-lg-4 col-md-5 col-sm-5 col-xs-12' }, [
 
-          div({ isRendered: darInfo.havePI, className: 'row no-margin' }, [
+          div({ isRendered: (researcherProfile.havePI === "true"), className: 'row no-margin' }, [
```
Comment (rushtong) [inline src/components/ApplicationSummary.js:27]: This was a bug, `havePI` is no longer on darInfo

File: src/components/ApplicationSummary.js [hunk 2]
```diff
@@ -23,14 +24,14 @@ export const ApplicationSummary = hh(class ApplicationSummary extends PureCompon
...
             label({ className: 'control-label access-color' }, ['PI: ']),
             span({ id: 'lbl_principalInvestigator', className: 'response-label', style: { 'paddingLeft': '5px' } },
               [darInfo.pi])
           ]),
           div({ className: 'row no-margin' }, [
             label({ className: 'control-label access-color' }, ['Researcher: ']),
-            span({ id: 'lbl_researcher', className: 'response-label', style: { 'paddingLeft': '5px' } }, [darInfo.profileName])
+            span({ id: 'lbl_researcher', className: 'response-label', style: { 'paddingLeft': '5px' } }, [researcherProfile.profileName])
```
Comment (rushtong) [inline src/components/ApplicationSummary.js:34]: This was a bug, `profileName` is no longer on darInfo

File: src/components/ApplicationSummary.js [hunk 3]
```diff
@@ -52,24 +53,24 @@ export const ApplicationSummary = hh(class ApplicationSummary extends PureCompon
           ]),
           div({ className: 'row no-margin' }, [
             label({ className: 'control-label access-color' }, ['Institution: ']),
-            span({ id: 'lbl_institution', className: 'response-label', style: { 'paddingLeft': '5px' } }, [darInfo.institution])
+            span({ id: 'lbl_institution', className: 'response-label', style: { 'paddingLeft': '5px' } }, [researcherProfile.institution])
```
Comment (rushtong) [inline src/components/ApplicationSummary.js:56]: This was a bug, `institution` is no longer on darInfo

File: src/components/ApplicationSummary.js [hunk 4]
```diff
@@ -52,24 +53,24 @@ export const ApplicationSummary = hh(class ApplicationSummary extends PureCompon
...
           ]),
           div({ className: 'row no-margin' }, [
             label({ className: 'control-label access-color' }, ['Department: ']),
-            span({ id: 'lbl_department', className: 'response-label', style: { 'paddingLeft': '5px' } }, [darInfo.department])
+            span({ id: 'lbl_department', className: 'response-label', style: { 'paddingLeft': '5px' } }, [researcherProfile.department])
```
Comment (rushtong) [inline src/components/ApplicationSummary.js:60]: This was a bug, `department` is no longer on darInfo

File: src/components/ApplicationSummary.js [hunk 5]
```diff
@@ -52,24 +53,24 @@ export const ApplicationSummary = hh(class ApplicationSummary extends PureCompon
...
           ]),
           div({ className: 'row no-margin' }, [
             label({ className: 'control-label access-color' }, ['City: ']),
-            span({ id: 'lbl_state', className: 'response-label', style: { 'paddingLeft': '5px' } }, [darInfo.city])
+            span({ id: 'lbl_state', className: 'response-label', style: { 'paddingLeft': '5px' } }, [researcherProfile.city])
```
Comment (rushtong) [inline src/components/ApplicationSummary.js:64]: This was a bug, `city` is no longer on darInfo

File: src/components/ApplicationSummary.js [hunk 6]
```diff
@@ -52,24 +53,24 @@ export const ApplicationSummary = hh(class ApplicationSummary extends PureCompon
...
           ]),
           div({ className: 'row no-margin' }, [
             label({ className: 'control-label access-color' }, ['Country: ']),
-            span({ id: 'lbl_country', className: 'response-label', style: { 'paddingLeft': '5px' } }, [darInfo.country])
+            span({ id: 'lbl_country', className: 'response-label', style: { 'paddingLeft': '5px' } }, [researcherProfile.country])
```
Comment (rushtong) [inline src/components/ApplicationSummary.js:68]: This was a bug, `country` is no longer on darInfo

File: src/pages/AccessResultRecords.js [hunk 7]
```diff
@@ -624,25 +615,21 @@ class AccessResultRecords extends Component {
       this.electionRPCall(electionId, false) //Election.findRPElectionReview(electionId, false (finalStatus))
     ]);
 
+    const datasets = await DataSet.getDarDatasets(darAPIResults.darInfo.datasetIds);
     const {darData, electionReview, showDULData, vaultVote} = electionReviewResults;
-    const restrictions = darAPIResults.darInfo.restrictions;
```
Comment (rushtong) [inline src/pages/AccessResultRecords.js:628]: unused

File: src/pages/AccessResultRecords.js [hunk 8]
```diff
@@ -624,25 +615,21 @@ class AccessResultRecords extends Component {
...
     const consent = electionReview.consent;
 
     assignToState = Object.assign({}, darData, electionReview, showDULData, vaultVote, electionAPIResults, darAPIResults, finalDACVote, electionRPResults);
-    assignToState.hasUseRestrictions = !isNil(restrictions) && !isEmpty(restrictions);
```
Comment (rushtong) [inline src/pages/AccessResultRecords.js:632]: unused, and always true

File: src/pages/AccessResultRecords.js [hunk 9]
```diff
@@ -624,25 +615,21 @@ class AccessResultRecords extends Component {
...
 
     //NOTE: lot of election calls, they seem to have different contexts and have extra stuff added to them
     //On top of that it just gets a bunch of other stuff (votes, consent) objects attached to it and returned as a seperate data structure
     //Contextually the difference between each of these elections are not clear (TYPE is different, but what that entails is not obvious)
     //Similar questions with the various votes (rpVotes, finalDACVote).
 
-    //NOTE: the component currently uses hasRestrictions, which I found out to simply be a check on the restrictions attribute on a DAR
-    //Basically all it does is check if a value exists and returns the bool value
-    //So all I've done is simply check that value myself from the describeDar return value rather than use the endpoint
-
```
Comment (rushtong) [inline src/pages/AccessResultRecords.js:645]: The original code was incorrectly testing if a restriction existed. A restriction **always** exists.

File: src/pages/AccessResultRecords.js [hunk 10]
```diff
@@ -655,7 +642,6 @@ class AccessResultRecords extends Component {
     //   election: electionReviewResults.showDULData.election,
     //   downloadUrl: electionReviewResults.showDULData.downloadUrl,
     //   dulName: electionReviewResults.showDULData.dulName,
-    //   hasUseRestrictions: !isNil(restrictions) && !isEmpty(restrictions),
```
Comment (rushtong) [inline src/pages/AccessResultRecords.js:658]: unused

File: src/pages/access_review/DarApplication.js [hunk 11]
```diff
@@ -26,11 +25,6 @@ const HEADER_BOLD = {
 
 export const DarApplication = hh(class DarApplication extends React.PureComponent {
 
-  downloadDAR = () => {
-    const { darId } = this.props;
-    Files.getDARFile(darId);
-  };
-
```
Comment (rushtong) [inline src/pages/access_review/DarApplication.js:33]: unused

**Review by raejohanek — APPROVED** [review]
These changes look good, awesome that you found all those bugs!

**Comment by rushtong** [comment]
Thank you @raejohanek!

**Pull request merged**